massa-execution-worker/src/interface_impl.rs

```diff
@@ -343,6 +343,27 @@ impl Interface for InterfaceImpl {
         Ok(context.has_data_entry(&addr, key))
     }
 
+    /// Check whether or not the caller has write access in the current context
+    ///
+    /// # Returns
+    /// true if the caller has write access
+    fn caller_has_write_access(&self) -> Result<bool> {
+        let context = context_guard!(self);
+        let mut call_stack_iter = context.get_call_stack().into_iter().rev();
+        let caller = if let Some(last) = call_stack_iter.next() {
```

Comment:
Something that I find strange is that you seems to fetch the right element in the stack here that is the caller address but after you are iterating again in `get_owned_addresses_for ` to find it again. Couldn't be possible to factorize them?

Reply:
True, I improved it